docs/api.adoc

```diff
@@ -7,58 +7,188 @@ v2.0.0, 2016-12
 
 Detailed description of api docs here!
 
-Hosts
+Search
 -----
-
 [nested]
-* List all
+* Search for host or group.
+Specify type as  a group, host, or any to limit the search.
 +
 [source]
 ....
-  /api/host/list
+GET /api/search?type=any&name=web_hosts
 ....
 ** Example results:
 +
 [source,json]
 ----
 {
+  "status": "success",
+  "code": 200,
+  "data": {
+  "groups": [
+    "webhosts1",
+    "webhosts2"
+    ],
   "hosts": [
-    "host1",
-    "host2",
-    "host9",
-    "host10"
+   "webhost1"
+   "webhost2"
   ]
 }
 ----
 
-* Search for specific host
+Hostvars
+--------
+
+[nested]
+* Retrieve a host and it's vars.
 +
 [source]
 ....
-/api/host/search?key=hostname
+GET /api/hostvars?name=host1
 ....
 ** Example results:
 +
 [source,json]
 ----
 {
-  "hosts": [
-    "host1",
-    "host10"
-  ]
+  "status": "successful",
+  "code": 200,
+  "data": {
+    "parent_group": "group1",
+    "host": "host1",
+    "vars": {
+      "foo": "bar",
+      "var": "1"
+      }
+    }
+}
+----
+
+* Retrieve a specific variable for that host.
++
+[source]
+....
+GET /api/hostvars?name=host1&var=foo
+....
+** Example results:
++
+[source,json]
+----
+{
+  "status": "successful",
+  "code": 200,
+  "data": {
+    "host": "host1",
+    "vars": {
+      "foo": "bar",
+      }
+    }
+}
+----
+
+* Add a new host
++
+[source]
+....
+POST /api/hostvars?name=newhostname
+....
+** Example response:
++
+[source,json]
+----
+{
+  "status": "successful",
+  "code": 201,
 }
 ----
+* Update or add a new var to a host.
+This call is idempotent, so if a value already matches no changes will be made.
++
+[source]
+....
+PUT /api/hostvars?name=newhostname&var1=1
+....
+** Example response:
++
+[source,json]
+----
+{
+  "status": "successful",
+  "code": 200
+}
+----
+
+
+* Delete a host
+[source]
+....
+DELETE /api/hostvars?name=name_of_the_host
+....
 
 Groups
 -----
+** Example response:
++
+[source,json]
+----
+{
+  "status": "successful",
+  "code": 204
+}
+----
 
 [nested]
-* List all
+* Get a group.
 +
 [source]
 ....
-/api/group/list
+GET /api/groupvars?name=group_name
 ....
+** Example response:
++
+[source,json]
+----
+{
+  "status": "successful",
+  "code": 200,
+  "data": {
+    "group_name": "group1",
+    "parent_group": "all",
```

Comment:
there will never be a parent_group called all.  for so called "top_level_groups", there will be a key called "parent_groups": []  assigned to an empty list.  

We should also show "children": [ ] here with a list of children groups.